servicetalk-http-api/src/main/java/io/servicetalk/http/api/HttpLifecycleObserver.java

```diff
@@ -117,26 +127,28 @@ default void onExchangeFinally() {
     /**
      * An observer interface that provides visibility into events associated with a single HTTP request.
      * <p>
-     * The request is considered complete when one of the terminal events is invoked. It's guaranteed only one terminal
+     * The request is considered complete when one of the terminal events is invoked. It is guaranteed only one terminal
      * event will be invoked per request.
      */
     interface HttpRequestObserver {
 
         /**
-         * Callback when subscriber requests {@code n} items of the request payload body.
+         * Callback when the subscriber requests {@code n} items of the request payload body.
          * <p>
-         * May be invoked multiple times. Helps to track when items are requested and when they are
-         * {@link #onRequestData(Buffer) delivered}.
+         * May be invoked multiple times and concurrently with other callbacks on this observer. Therefore, it should
+         * have its own isolated state or should be synchronized if the state is shared with other callbacks.
+         * It can help track when items are requested and when they are {@link #onRequestData(Buffer) delivered}.
```

Comment:
It can help `to` track ..?

Reply:
I originally had `to` there, but AI proofreading dropped it 😄 
Asked a different model again:
> Both "It can help track" and "It can help to track" are grammatically correct and convey the same meaning. The addition of "to" before "track" is a stylistic choice and doesn't change the fundamental meaning. In most cases, you can use either phrase interchangeably.

Reply:
I added it back 😄